src/dbnode/storage/database.go

```diff
@@ -229,12 +230,28 @@ func NewDatabase(
 			zap.Error(err))
 	}
 
-	mediator, err := newMediator(
+	d.mediator, err = newMediator(
 		d, commitLog, opts.SetInstrumentOptions(databaseIOpts))
 	if err != nil {
 		return nil, err
 	}
-	d.mediator = mediator
+
+	d.repairer = newNoopDatabaseRepairer()
+	if opts.RepairEnabled() {
+		d.repairer, err = newDatabaseRepairer(d, opts)
+		if err != nil {
+			return nil, err
+		}
+		d.mediator.RegisterBackgroundProcess(d.repairer)
```

Comment:
This is slightly different from existing functionality; currently the `newNoopDatabaseRepairer` gets set if RepairEnabled is false

Reply:
Not sure I understand: do you want this `noopDatabaseRepairer` to be added to the list of background processes so that it actually gets executed?

Reply:
Ah cool missed that it was set on the db above